Add info about new contraction data type and permutation feature to README (#167)
ROCm/hipTensor: README.md

* AMD CDNA class GPU featuring matrix core support:
gfx908, gfx90a, gfx940, gfx941, gfx942 as 'gfx9'

:::{note}
Double precision FP64 datatype support requires gfx90a, gfx940, gfx941 or gfx942
:::

> ## Minimum Software Requirements

Reviewer: Can we make sure all titles have sentence case?
```suggestion
## Minimum software requirements
```

## Currently supported

> ### Operations - Contraction Tensor

Reviewer: Does "Contractor Tensor" need to be capitalized (i.e. is it a formal name)? All titles should be in sentence case.

> ### Operations - Contraction Tensor

Reviewer: ```suggestion
### Operation: Contraction tensor
```

hiptensor supports a tensor contraction of the form $D = \alpha AB + \beta C$

Supported data-type combinations are:

| typeA       | typeB       | typeC       | typeCompute       | notes                              |
| ----------- | ----------- | ----------- | ----------------- | ---------------------------------- |
| bf16        | bf16        | bf16        | f32               |                                    |
| __half      | __half      | __half      | f32               |                                    |
| f32         | f32         | f32         | bf16              |                                    |
| f32         | f32         | f32         | __half            |                                    |
| f32         | f32         | f32         | f32               |                                    |
| f64         | f64         | f64         | f32               | f64 is supported only on gfx90a +  |
| f64         | f64         | f64         | f64               | f64 is supported only on gfx90a +  |

> ### Operations - Permutation Tensor

Reviewer: ```suggestion
### Operation: Permutation tensor
```

## Currently supported

### Operation: Contraction tensor

hiptensor supports a tensor contraction of the form $D = \alpha AB + \beta C$

Supported data-type combinations are:

| typeA       | typeB       | typeC       | typeCompute       | notes                              |
|-------------|-------------|-------------|-------------------|------------------------------------|

> | bf16        | bf16        | bf16        | f32               |                                    |

Reviewer: Should the letters in this table not be in caps? (e.g., BF16)

Author: I think we can keep the bf16 to be consistent with the RocWMMA's format. 

RocWMMA's readme https://github.com/ROCm/rocWMMA

![image](https://github.com/ROCm/hipTensor/assets/142121551/ca664024-8689-4845-8c6a-96e993470fd5)

| __half      | __half      | __half      | f32               |                                    |
| f32         | f32         | f32         | bf16              |                                    |
| f32         | f32         | f32         | __half            |                                    |
| f32         | f32         | f32         | f32               |                                    |

> | f64         | f64         | f64         | f32               | f64 is supported only on gfx90a +  |

Reviewer: I would suggest changing "is supported only on" to "is only supported on".

Author: changed to "is only supported on".

| HIPTENSOR_BUILD_TESTS   | Build Tests                           | ON                                                             |
| HIPTENSOR_BUILD_SAMPLES | Build Samples                         | ON                                                             |

### Example configurations

By default, the project is configured as Release mode.
Here are some of the examples for the configuration:
| Configuration                    | Command                                                                   |
|----------------------------------|---------------------------------------------------------------------------|
| Basic                            | `CC=hipcc CXX=hipcc cmake -B<build_dir> .`                                |
| Targeting gfx908                 | `CC=hipcc CXX=hipcc cmake -B<build_dir> . -DAMDGPU_TARGETS=gfx908:xnack-` |
| Debug build                      | `CC=hipcc CXX=hipcc cmake -B<build_dir> . -DCMAKE_BUILD_TYPE=Debug`       |
| Build without tests (default on) | `CC=hipcc CXX=hipcc cmake -B<build_dir> . -DHIPTENSOR_BUILD_TESTS=OFF`    |

> After configuration, build with `cmake --build <build_dir> -- -j<nproc>`

Reviewer: ```suggestion
After configuration, build with `cmake --build <build_dir> -- -j<nproc>`.
```

* Bilinear contraction tests

Tests the API implementation of bilinear contraction algorithm with validation.

```bash
  <build_dir>/bin/bilinear_contraction_test
```

* Scale contraction tests

Tests the API implementation of scale contraction algorithm with validation.

```bash
  <build_dir>/bin/scale_contraction_test
```

## Running permutation tests

Test API implementation of the permutation algorithm with validation.

```bash
  <build_dir>/bin/permutation_test
```

## Samples

These are stand-alone use-cases of the hipTensor contraction operations.

### F32 bilinear contraction

Demonstrates the API implementation of bilinear contraction operation without validation.

```bash
  <build_dir>/bin/simple_bilinear_contraction_<typeA>_<typeB>_<typeC>_<typeD>_compute_<computeType>
```

### F32 scale contraction

Demonstrates the API implementation of scale contraction operation without validation.

```bash
  <build_dir>/bin/simple_scale_contraction_<typeA>_<typeB>_<typeD>_compute_<typeCompute>
```

### Permutation

Demonstrates the API implementation of permutation operation without validation.

```bash
  <build_dir>/bin/simple_permutation
```

### Build Samples as external client

> The client application links to hipTensor library; therefore, you must install the

Reviewer: ```suggestion
The client application links to the hipTensor library; therefore, you must install the
```

### Logger tests

Tests API implementation of logger verbosity and functionality.

```bash
  <build_dir>/bin/logger_test
```

> ## Running Contraction Tests

Reviewer: ```suggestion
## Running contraction tests
```

> ### Bilinear contraction tests

Reviewer: Can we make this heading (and Scale contraction tests) into bullet points in order to avoid stacked headings?

Tests the API implementation of bilinear contraction algorithm with validation.

```bash
  <build_dir>/bin/bilinear_contraction_test
```

### Scale contraction tests

Tests the API implementation of scale contraction algorithm with validation.

```bash
  <build_dir>/bin/scale_contraction_test
```

> ## Running Permutation Tests

Reviewer: This should be sentence case:

```suggestion
## Running permutation tests
```

> ### Permutation tests

Reviewer: I don't think we need this heading.

> Tests API implementation of permutation algorithm with validation.

Reviewer: ```suggestion
Test API implementation of the permutation algorithm with validation.
```

```bash
  <build_dir>/bin/permutation_test
```

## Samples

These are stand-alone use-cases of the hipTensor contraction operations.

> ### F32 Bilinear contraction

Reviewer: ```suggestion
### F32 bilinear contraction
```

Demonstrates the API implementation of bilinear contraction operation without validation.

```bash
  <build_dir>/bin/simple_bilinear_contraction_<typeA>_<typeB>_<typeC>_<typeD>_compute_<computeType>
```

> ### F32 Scale contraction

Reviewer: ```suggestion
### F32 scale contraction
```

Demonstrates the API implementation of scale contraction operation without validation.

```bash
  <build_dir>/bin/simple_scale_contraction_<typeA>_<typeB>_<typeD>_compute_<typeCompute>
```

### Permutation

Demonstrates the API implementation of permutation operation without validation.

```bash
  <build_dir>/bin/simple_permutation
```

### Build Samples as external client

> Client application links to hipTensor library, and therefore hipTensor library

Reviewer: How about: "The client application links to hipTensor library; therefore, you must install the hipTensor library before building client applications."

```

## Running contraction tests

* Bilinear contraction tests

Tests the API implementation of bilinear contraction algorithm with validation.

```bash
  <build_dir>/bin/bilinear_contraction_test
```

* Scale contraction tests

Tests the API implementation of scale contraction algorithm with validation.

```bash
  <build_dir>/bin/scale_contraction_test
```

## Running permutation tests

Test API implementation of the permutation algorithm with validation.

```bash
  <build_dir>/bin/permutation_test
```

## Samples

These are stand-alone use-cases of the hipTensor contraction operations.

### F32 bilinear contraction

Demonstrates the API implementation of bilinear contraction operation without validation.

```bash
  <build_dir>/bin/simple_bilinear_contraction_<typeA>_<typeB>_<typeC>_<typeD>_compute_<computeType>
```

### F32 scale contraction

Demonstrates the API implementation of scale contraction operation without validation.

```bash
  <build_dir>/bin/simple_scale_contraction_<typeA>_<typeB>_<typeD>_compute_<typeCompute>
```

### Permutation

Demonstrates the API implementation of permutation operation without validation.

```bash
  <build_dir>/bin/simple_permutation
```

> ### Build Samples as external client

Reviewer: ```suggestion
### Build samples as external client
```
